rewrite-java/src/main/java/org/openrewrite/java/RemoveUnusedImports.java

```diff
@@ -297,6 +297,17 @@ public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionCon
 
             return cu;
         }
+
+        private static boolean conflictsWithJavaLang(J.Import elem) {
+            try {
+                ClassLoader.getSystemClassLoader().loadClass("java.lang." + elem.getClassName());
```

Comment:
Not really a fan of checking for existence of classes through the System classloader; do we have any better alternatives?
There's quite some classes in `java.lang` https://docs.oracle.com/javase/8/docs/api/java/lang/package-summary.html that might conflict, although I'd expect such cases to be rare, and the list not to grow much.

Should we instead use a collection here of existing class names for a quick & safe way to filter?

Reply:
Agreed. Using the class loader doesn't make much sense, since that doesn't match the target Java version of the source code being modified.